DUI3-326:  Add civil3d 2024 converter

clairekuang opened the pull request:
Adds the Civil3d 2024 converter.

This pr includes **required changes to container registration** in order to support converter injection into a connector vertical.
These changes should be reflected in the other DUI3 converter DI projects before this pr is merged.

First pipe sent to speckle from Civil3d: https://app.speckle.systems/projects/b53a53697a/models/719612f9da

## Description & motivation


## Changes:

- sdk container registration
- missing autocad conversions
- civil3d 2024 converter 

## To-do before merge:

- [x] Fix container registration for all other DUI3 converter DI projects

oguzhankoral reviewed (changes requested):
Only reason requesting change on static class, but not sure it was a hack for multiple converter registration?
Q: why we removed ITypedConversions and extracted them? In order to use them on Civil3D too?

adamhathcock commented on DUI3-DX/Converters/Autocad/Speckle.Converters.Autocad2024.DependencyInjection/AutocadConverterModule.cs:
```diff
@@ -5,7 +5,7 @@
 using Speckle.Converters.Common.DependencyInjection;
 
-namespace Speckle.Converters.Autocad20243.DependencyInjection;
+namespace Speckle.Converters.Autocad2024.DependencyInjection;
```
I think it's a tester for when we do.

clairekuang replied:
Yes the autocad 2024 converter is required for civil3d 2024

adamhathcock commented on DUI3-DX/Converters/Autocad/Speckle.Converters.AutocadShared/ToSpeckle/Raw/DBCurveToSpeckleRawConverter.cs:
```diff
@@ -0,0 +1,67 @@
+
+namespace Speckle.Converters.Autocad.ToSpeckle.Raw;
+
+public class DBCurveToSpeckleRawConverter : ITypedConverter<ADB.Curve, Objects.ICurve>, ITypedConverter<ADB.Curve, Base>
```
shouldn't be any different than just 1

clairekuang commented:
> Q: why we removed ITypedConversions and extracted them? In order to use them on Civil3D too?

There is no difference between having a converter implement both `ITopLevelConverter` and `ITypedConverter`, or to separate out the typed converter class to be called from the top level converter. I thought the second option is cleaner, so we can see at a glance all our typed converters.
